### DISCUSSION TITLE
Which dialog::choice* should be used?

### DISCUSSION THREAD

**Discussion by mark-summerfield** [fltk-rs/fltk-rs#1157] [Q&A] [answered]
fltk-rs offers these functions: dialog::choice(), dialog::choice2(), dialog::choice2_default(), and dialog::choice_default(). But there doesn't seem to be any explanation of what the differences are or whether any are preferred or any deprecated.

**Comment by MoAlyousef** [comment, marked as answer]
Hi
This needs better documentation. So dialog::choice() and choice_default() only differ in the arguments being passed to the function, *_default() functions don't take an x,y coordinate, and the dialog is shown where the pointer currently is.

choice vs choice2:
choice2 wraps the newly added fl_choice_n() which actually returns a negative value indicating no choice was selected. (the dialog was closed). Similarly choice2 returns a None when the dialog is closed while nothing was selected. Whereas choice() returns 0 whether the first choice is selected or the dialog window was closed.

I'm thinking of deprecating choice in favor of choice2, then changing the name with version 2.0.
